src/routes/recovery/recovery.service.ts

```diff
@@ -0,0 +1,26 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { IAlertsRepository } from '@/domain/alerts/alerts.repository.interface';
+import { AlertsRepository } from '@/domain/alerts/alerts.repository';
+import { AddRecoveryModuleDto } from '@/routes/recovery/entities/add-recovery-module.dto.entity';
+import { Contract } from '@/domain/alerts/entities/alerts.entity';
+
+@Injectable()
+export class RecoveryService {
+  constructor(
+    @Inject(IAlertsRepository)
+    private readonly alertsRepository: AlertsRepository,
+  ) {}
+
+  async addRecoveryModule(args: {
+    chainId: string;
+    safeAddress: string;
+    addRecoveryModuleDto: AddRecoveryModuleDto;
+  }): Promise<void> {
+    const contract: Contract = {
+      chainId: args.chainId,
+      address: args.addRecoveryModuleDto.moduleAddress,
+      displayName: `${args.chainId}:${args.safeAddress}`,
+    };
```

Comment:
I think having `Contract` for the alert entity can be confusing as we already have a domain entity named `Contract` which is our representation of a `Contract`.

Maybe this can one can be associated with alerts directly and we can call it `AlertsRegistration` or similar? What do you think?

We can tackle it in a different PR as it affects the domain part of Alerts.

Reply:
I agree. I've made a note to change it in a follow-up PR.